tls/s2n_psk.c

```diff
@@ -474,15 +500,9 @@ int s2n_connection_append_psk(struct s2n_connection *conn, struct s2n_psk *input
     }
 
     DEFER_CLEANUP(struct s2n_psk new_psk = { 0 }, s2n_psk_wipe);
-    ENSURE_POSIX(s2n_result_is_ok(s2n_psk_init(&new_psk, S2N_PSK_TYPE_EXTERNAL)), S2N_ERR_INVALID_ARGUMENT);
-    ENSURE_POSIX(s2n_psk_set_identity(&new_psk, input_psk->identity.data, input_psk->identity.size) == S2N_SUCCESS,
-            S2N_ERR_INVALID_ARGUMENT);
-    ENSURE_POSIX(s2n_psk_set_secret(&new_psk, input_psk->secret.data, input_psk->secret.size) == S2N_SUCCESS,
-            S2N_ERR_INVALID_ARGUMENT);
-    new_psk.hmac_alg = input_psk->hmac_alg;
-
+    ENSURE_POSIX(s2n_result_is_ok(s2n_psk_clone(&new_psk, input_psk)), S2N_ERR_INVALID_ARGUMENT);
```

Comment:
Just making sure you're wanting to hide any errors in the underlying clone function.

Reply:
I think it's the right choice. Cloning issues would be triggered by problems with the input_psk, hence S2N_ERR_INVALID_ARGUMENT.